### PR TITLE
Serialize and deserialize dates using ISO format

### PR DESCRIPTION
**Pull request by pushplay**
Dates serialize to ISO format.  ISO date string deserialize to Dates when `dynamicTyping` is true.

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -408,6 +408,9 @@
 			if (typeof str === 'undefined' || str === null)
 				return '';
 
+			if (str.constructor === Date)
```
Comment (pushplay) [inline papaparse.js]: This comparison is faster than instanceof but won't work for subclasses of Date.  If someone is weird enough to extend Date they can modify toString() to get their desired behavior.

Comment (pokoli) [reply papaparse.js]: I think I will prefer to not overoptimize and use instanceof and make thinks works as expected.

File: tests/test-cases.js [hunk 2]
```diff
@@ -1422,6 +1431,11 @@ var UNPARSE_TESTS = [
 		input: [{"Col1": "a", "Col2": "b", "Col3": "c"}, {"Col1": "d", "Col2": "e", "Col3": "f"}],
 		config: {header: false},
 		expected: 'a,b,c\r\nd,e,f'
+	},
+	{
+		description: "Date handling",
```
Comment (pokoli) [inline tests/test-cases.js]: I think this description can be improved

**Comment by pokoli** [comment]
I've fixed conflics and test description and merged this manually on 757dc379151580de9fd1022254ceadb4b7944d6b

Thanks for your contribution.

**Pull request closed without merging**